### DISCUSSION TITLE
The constraints order of "export" function

### DISCUSSION THREAD

**Discussion by BYS543** [yalmip/YALMIP#922] [Problems using YALMIP] [answered]
Hi, Johan
If a model has the equlity and inequlity constraints, I find the equlity constraints will be ranked at the front row when using "export".
If the constraints set is as follows,
`Constraints=[x(1)+x(2)==1,x(1)+x(3)<=3,x(2)<=1,x(2)+x(5)==2];`
is the ranked constraints set when using "export" like:
`Constraints=[,x(1)+x(2)==1,x(2)+x(5)==2,x(1)+x(3)<=3,x(2)<=1];`
Thanks in advance!

**Comment by johanlofberg** [comment, marked as answer]
Well, it depends on which solver you are exporting the data to. The internal numerical format, which is an extension of Sedumis, packs data as equality/lp/socp cone/exp cone/power cone/sdp cone with the size of each defined by the K.f/l/q/e/p/s fields

**Reply by BYS543** [reply]
Thank you,Johan. How to understand the meaning of K.f/l/q/e/p/s. I don't really know the format of Sedumi. If the solver I used is Gurobi, do constraints of the same type(e.g. inequality) always follow the principle of occurrence order?

**Reply by johanlofberg** [reply]
f/l/e are scalars with the number of those constraints, while q/p/s are vectors with the size of each individual corresponding cone constraint

However, if you intend to export to some solver format, there is no reason to bother with the internal format. The particulars for a specific solver you would have to look in the solver documentation (and simply look at the exported data, sometimes it is obvious with matrices called model.Aeq etc. Just export a trivially identifiable model)
